idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/intention/IntentionSyntax.kt

```diff
@@ -79,7 +75,7 @@ interface IntentionSyntax : IntentionUtilitySyntax {
     text: String = "",
     kClass: Class<K> = KtElement::class.java as Class<K>,
     isApplicableTo: (element: K, caretOffset: Int) -> Boolean = Noop.boolean2False,
-    applyTo: (element: K, editor: Editor?) -> Unit = Noop.effect2,
+    applyTo: KtPsiFactory.(element: K, editor: Editor) -> Unit = Noop.effect3,
```

Comment:
I don't fully understand the need for a `KtPsiFactory` as receiver here. 

This is always called with a new instance (via `it.project?.ktPsiFactory?` below) and it might be unused by the callers of this method `IntentionSyntax.ktIntention`.
Can't callers of `IntentionSyntax.ktIntention` just create their own instance when needed? 

If it's required and necessary to pass as a receiver, then you might want to cache the instance. For example, with a key attached to the project (which is a UserDataHolder). Something like this:
```
val KEY = Key.create<KtPsiFactory>("arrow.ktpsifactory")
KEY.set(project, KtPsiFactory(project))
// later
val ktPsiFactory = KEY.get(project)
```

Reply:
They can, but `KtPsiFactory` has a lot of utilities users will end up using, when they're manipulating the intercepted KtElement.
IMO there is no need to cache the instance unless we have sever performance drawbacks.
And I could detect them, yet, as we haven't used a lot of Intentions or Inspections in our project.

Reply:
I agree about the caching.
I still don't entirely agree about the `KtPsiFactory`, though. There's a lot of other, potentially useful  classes which may be used by the users of this method. This may be my personal opinion about style of apis. In the end it's your call what's best here :)

Reply:
@jansorg Do you have suggestions which other APIs may be interesting for users? We can add them.
Nothing prevents the user to use other APIs or classes manually, but we can also provide them as top-level constructs/ services in each function so that users have direct access without researching for rather `common` undocumented pit-holes.

Reply:
I wanted to say that assumptions like this might be too smart for a generic api :)
This method assumes that the user will certainly need a `KtPsiFactory`. Some will need it, some won't. Some users will need a `com.intellij.psi.PsiElementFactory` for Java, some will need both factories, other will need something completely different.
(And a reference to `KtPsiFactory` requires a hard dependency on the Kotlin plugin.)

Personally, I'd probably add a note in the method's documentation to educate the user instead of always creating a new instance.